### PR TITLE
[TCES-18] Connecting Supabase database to the current codebase from the old PlanetScale database

### PR DESCRIPTION
**Pull request by williamntlam**
# Description

Finished setting up the new Supabase database to the current codebase. In this commit (https://github.com/uoftblueprint/tces/commit/d05aebfd36ea2ba827d354a273c66a26f248c28d), I ran all the provided unit tests and they all passed. I also wrote some new unit tests to test uploading a text file and a PDF file onto the Supabase S3 bucket and it seems to work. I hope that helps.

William.

### CODE REVIEW

**Comment by 04kash** [comment]
Workflows probably need to be updated to use this new database.

**Comment by williamntlam** [comment]
Just an update:

I'm testing that everything on the client side still works with the new connection to the Supabase database. Right now, there are some issues with some schemas involving setting default values with foreign key attributes.

Also, the CI currently isn't passing. This is most likely due to the .env variables used for the new database. This needs to be fixed too.

William.

**Comment by williamntlam** [comment]
Another update:

I managed to fix some bugs but there are still remaining bugs in the code

For the client side:
- When the main dashboard first loads after logging in, there seems to be an error. But, refreshing it seems to fix it.
- When a new client is created, the user is stuck in the "Create User" tab and has to exit out of it. The new client is successfully created.

For the server side:
- In the Clients tab, under the timeline feature, add new contact does not work.
- For the Employees tab, under the timeline feature, fetching jobs does not work (Creating a job might not work too - I forgot whether I tested it or not unfortunately).
- Applying and resetting search filters for Clients results in an error.

Hope that helps!

William.

File: server/index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const https = require("https");
 const fs = require("node:fs");
 
 const app = express();
-const port = 8000;
+const port = 8001;
```
Comment (04kash) [inline server/index.js:11]: Just wondering why was changed to 8001?
This would require changes to the docker-compose.yml as well as .env file in `client`

Comment (williamntlam) [reply server/index.js:11]: Oh I forgot to change it back. I was testing it and I was getting "Port 8000 was being used" errors and I was impatient so I just changed it to 8001 while testing. I think it's okay to change it back to Port 8000. Sorry!

**Comment by 04kash** [comment]
Closing this as we'll be using AWS RDS now as discussed.

**Pull request closed without merging**